edge/pkg/edgehub/clients/factory_test.go

```diff
@@ -59,15 +60,16 @@ func TestGetClient(t *testing.T) {
 			ReadDeadline:     100 * time.Second,
 			ExtendHeader:     http.Header{},
 		}),
+			nil,
 		},
 
 		{"TestGetClient: Negative Test Case", args{
 			clientType: "WrongClientType",
-		}, nil},
+		}, nil, ErrorWrongClientType},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := GetClient(tt.args.clientType, tt.args.config); !reflect.DeepEqual(got, tt.want) {
+			if got, err := GetClient(tt.args.clientType, tt.args.config); !reflect.DeepEqual(got, tt.want) || err != tt.err {
```

Comment:
I think we can split the error checks but I don't want to block this PR too long. We can improve it in follow-ups.

Reply:
ok